osu.Game/Skinning/SkinLayoutInfo.cs

```diff
@@ -19,12 +19,26 @@ public class SkinLayoutInfo
     {
         private const string global_identifier = @"global";
 
-        [JsonIgnore]
-        public IEnumerable<SerialisedDrawableInfo> AllDrawables => DrawableInfo.Values.SelectMany(v => v);
+        /// <summary>
+        /// Latest version representing the schema of the skin layout.
+        /// </summary>
+        /// <remarks>
+        /// <list type="bullet">
+        /// <item><description>0: Initial version of all skin layouts.</description></item>
+        /// <item><description>1: Moves existing combo counters from global to per-ruleset HUD targets.</description></item>
+        /// </list>
+        /// </remarks>
+        public const int LATEST_VERSION = 1;
+
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Populate)]
+        public int Version = LATEST_VERSION;
```

Comment:
I don't know if I'm approaching this correctly but I'm using `DefaultValueHandling.Populate` so that when deserialising any skin prior to this PR, the value of `Version` will be assigned to `0` (default int value) instead of `LATEST_VERSION`.

Reply:
The documentation on this is [hilariously bad:](https://www.newtonsoft.com/json/help/html/t_newtonsoft_json_defaultvaluehandling.htm)

> Members with a default value but no JSON will be set to their default value when deserializing.

Like that sounds like the *inverse* to the actual desired behaviour here. That said, as far as my real-world testing is concerned, this _does_ appear to function as advertised...